src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala

```diff
@@ -40,13 +40,14 @@ trait RouteHandling {
 
   /**
     * Directive handling validation errors as bad requests(400)
+    * @param entity entity on which query should be ran
     * @param fieldQuery field query before validation
     * @return Either validated query or bad request if validation failed
     */
-  protected def validateQueryOrBadRequest(fieldQuery: Query): Directive1[Query] = Directive { inner =>
-    fieldQuery.validate match {
-      case Some(value) => inner(Tuple1(value))
-      case None => complete(StatusCodes.BadRequest)
+  protected def validateQueryOrBadRequest(entity: String, fieldQuery: ApiQuery): Directive1[Query] = Directive { inner =>
+    fieldQuery.validate(entity) match {
+      case Right(value) => inner(Tuple1(value))
+      case Left(errors) => complete(StatusCodes.BadRequest -> s"Errors: ${errors.mkString(",")}")
```

Comment:
Did we consider returning a json content to parse even when there are errors?

Reply:
Possibly using `mkString` with a `\n` separator might be more readable as text

Reply:
ok, I've added `\n`